examples/inspection/plot_model_inspection_cross_validation.py

```diff
@@ -0,0 +1,229 @@
+
+# %%
+# We see that, even if there are several values far beyond `alpha=40`, most of
+# the models tuned within the cross-validation had a regularization parameter
+# around this value. This is a good sign that this choice is stable across
+# folds.
+#
+# **Further inspection of linear models**
+#
+# Not only the hyperparameters such as `alpha` should be studied. If a model has
+# coefficients coming out of the fitting process, they should be analyzed as
+# well. In our case, we used a linear model. These models are parametrized and
+# rely on two parameters: `coef_` and `intercept_`. Therefore, we should analyze
+# the variance of these parameters as well.
+#
+# For the sake of simplicity, we are going to solely look at the `coef_`.
+import pandas as pd
+
+coefficients = pd.DataFrame(
+    [pipeline["ridge_regressor"].coef_ for pipeline in cv_pipelines],
+    columns=X.columns,
+)
+coefficients
+
+# %%
+import seaborn as sns
+
+plt.figure(figsize=(9, 7))
+sns.stripplot(data=coefficients, orient="h", palette="dark:k", alpha=0.5)
+sns.boxplot(data=coefficients, orient="h")
+plt.axvline(x=0, color=".5")
+plt.title("Coefficient values our model")
+_ = plt.subplots_adjust(left=0.3)
+
+# %%
+# We observe that the coefficients do not vary much taking the size of the
+# dataset into account, meaning that all the trained models are similar. Each
+# individual model is expected to more or less give the same predictions.
+#
+# This would not necessarily have been the case if the number of training
+# samples had been smaller, if the model had not been properly regularized or
+# if we had strongly dependent features in the data.
+#
+# Finalizing the predictive model
+# --------------------------------
+#
+# In the above analysis, we saw that the internally tuned value of `alpha`
+# did not vary much across cross-validation splits.
+# Indeed, we can retrain this model and optimize the value of `alpha` on the
+# full training set.
+from sklearn.base import clone
+
+final_model = clone(model)
+final_model.fit(X_train, y_train)
+
+# %%
+# At the beginning of the process, when we split our data, we had a set of left
+# out data. Now, we can use it to further check if the model performs as we
+# would expect from the analysis done within the cross-validation framework.
+print(f"The performance of our final model: R2={final_model.score(X_test, y_test):.2f}")
+
+# %%
+# We see that the statistical performance is comparable to the cross-validation
+# statistical performances which is not surprising. Similarly, we could look at
+# the coefficients of the final model and compare them with the coefficients
+# obtained within the cross-validation study.
+#
+# However, you should be aware that this latest step does not give any
+# information about the variance of our final model. Thus, if we want to
+# evaluate our final model, we should get a distribution of scores if we would
+# like to get this information.
```

Comment:
What is "variance of our final model". Do you have the bias variance decomposition in mind?
How would we get the "distribution of scores"?
I'm inclined to remove this remark.